### PR TITLE
Hide secret cloudformation param

### PR DESCRIPTION
**Pull request by DavidLawes**
## What does this change?
It was identified that some cloudformation params were considered secret so should be hidden on the aws console. Adding NoEcho: true will ensure that this value is hidden when viewing resources via the aws console.

While reviewing this repo it was found that these lambdas were built using travis, not team city. For the frontend-lambda repo a team city project has been created with a build step to deploy the facia purger. Future lambdas would require an additional build step definition.

Lots of changes here as a result of testing, intend to squash on merge.

## How to test
Before change, for facia-purger cloudformation stack, we can view the fastly api key:
Screenshot 2021-08-31 at 14 10 56

After the change, the fastly api key is hidden:
Screenshot 2021-09-02 at 12 01 34

Note:
To update cloudformation stacks a change set is created. A change set represents the difference between the existing cloudformation and the new one provided. In order to deploy an updated cloudformation template the change set must show some changes. If there are no changes identified then in riff-raff the cloudformation deploy step is successful and no changes are made (riff-raff continues then to deploy the lambda code).

In the cloudformation template we only wanted to add the NoEcho property to a parameter. This addition was not recognised as a change in the change set, so an automated deploy via riff-raff did not result in an updated cloudformation template. Also manually via the aws console, just updating the template directly didn't register a change. 

In order to update the cloudformation stack to include the NoEcho property a 2-step process was needed:
- update the template with the NoEcho property and also another (trivial change)
- update the template with the trivial change reverted

### CODE REVIEW

**Comment by DavidLawes** [comment]
[Cloudformation stacks that appear to be secrets but do not have the NoEcho flag](https://trello.com/c/Nqk2e5m0/240-cloudformation-stacks-that-appear-to-be-secrets-but-do-not-have-the-noecho-flag)

**Review by nicl — COMMENTED** [review]
I wonder if the TC setup is a bit confusing atm? Should we call is facia-purge-lambda as the only build step is for that?

And then, if we wanted to later, we could create a separate TC build for the backup parameter store lambda?

**Comment by DavidLawes** [comment]
> I wonder if the TC setup is a bit confusing atm? Should we call is facia-purge-lambda as the only build step is for that?
> 
> And then, if we wanted to later, we could create a separate TC build for the backup parameter store lambda?

Is it the naming that you think is confusing? Or the build step? Or something else?

About naming, I thought the most intuitive thing was to name the TC project after the github repo (to me this is the thing I would search by if trying to find a project...)

My understanding was that this was intended to be a multi-lambda repo. If we remove the other lambda then I'd be tempted to re-name the repo and restructure it so it's clear that only the facia purger is included (especially given that I don't _think_ this repo holds all the lambdas for frontend in the way it was perhaps intended). Then I think renaming the team city project would make sense too.

**Comment by nicl** [comment]
@DavidLawes yeah agree it's a sort of hybrid atm, which is confusing. Ideally, we'd add (separately/later) CI/D for the backup parameters lambda too. And I think a separate TC build would make sense for that - because it doesn't make sense to rebuild both when changing just one (as there is no shared code). So at that point, repo-based naming wouldn't really work.

**Comment by DavidLawes** [comment]
> @DavidLawes yeah agree it's a sort of hybrid atm, which is confusing. Ideally, we'd add (separately/later) CI/D for the backup parameters lambda too. And I think a separate TC build would make sense for that - because it doesn't make sense to rebuild both when changing just one (as there is no shared code). So at that point, repo-based naming wouldn't really work.

Updated team city project name to `facia-purger-lambda`:

Screenshot 2021-09-06 at 10 10 03

File: README.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
 # frontend-lambda
 
 frontend-lambda contains various self-contained AWS lambda projects used by frontend
+
+## How to deploy
```
Comment (jlieb10) [inline README.md:5]: 👍

**Review by jlieb10 — APPROVED** [review]
Nice work

**Pull request merged**